core-server/src/main/java/org/glassfish/jersey/server/ServerRuntime.java

```diff
@@ -510,7 +522,7 @@ private Response mapException(final Throwable originalThrowable) throws Throwabl
                             && throwable.getCause() instanceof MessageBodyProviderNotFoundException) {
                         throw throwable;
                     }
-                    Response waeResponse;
```

Comment:
Now we probably can move the HeaderValueException block back where it was in 3.0.3 so that HeaderValueException can be mappable.